backend/vnc.c

```diff
@@ -0,0 +1,268 @@
+        tev.u.pointer.screen_x = x;
+        tev.u.pointer.screen_y = y;
+        tev.kind = TwinEventButtonDown;
+        tev.u.pointer.button = 1;
+    } else if (!(button & NVNC_BUTTON_LEFT) &&
+               (peer->prev_button & NVNC_BUTTON_LEFT)) {
+        tev.u.pointer.screen_x = x;
+        tev.u.pointer.screen_y = y;
+        tev.kind = TwinEventButtonUp;
+        tev.u.pointer.button = 1;
+    }
+    if ((peer->px != x || peer->py != y)) {
+        peer->px = x;
+        peer->py = y;
+        tev.u.pointer.screen_x = x;
+        tev.u.pointer.screen_y = y;
+        tev.u.pointer.button = 0;
+        tev.kind = TwinEventMotion;
+    }
+    peer->prev_button = button;
+    struct nvnc *server = nvnc_client_get_server(client);
+    twin_vnc_t *tx = nvnc_get_userdata(server);
+    twin_screen_dispatch(tx->screen, &tev);
+}
+
+twin_context_t *twin_vnc_init(int width, int height)
+{
+    twin_context_t *ctx = calloc(1, sizeof(twin_context_t));
+    if (!ctx)
+        return NULL;
+    ctx->priv = calloc(1, sizeof(twin_vnc_t));
+    if (!ctx->priv) {
+        free(ctx);
+        return NULL;
+    }
+
+    twin_vnc_t *tx = ctx->priv;
+    tx->width = width;
+    tx->height = height;
+
+    tx->aml = aml_new();
+    if (!tx->aml) {
+        log_error("Failed to create aml");
+        goto bail_priv;
+    }
+    aml_set_default(tx->aml);
+    char *vnc_host = getenv(NEATVNC_HOST);
+    if (!vnc_host) {
+        log_info(
+            "Environment variable $NEATVNC_HOST not set, use %s by default",
+            NEATVNC_HOST_DEFAULT);
+        vnc_host = NEATVNC_HOST_DEFAULT;
+    }
+    char *vnc_port = getenv(NEATVNC_PORT);
+    if (!vnc_port) {
+        log_info(
+            "Environment variable $NEATVNC_PORT not set, use %s by default",
+            NEATVNC_PORT_DEFAULT);
+        vnc_port = NEATVNC_PORT_DEFAULT;
+    }
+    log_info("NeatVNC server IP %s PORT %s", vnc_host, vnc_port);
+    tx->server = nvnc_open(vnc_host, atoi(vnc_port));
+    if (!tx->server) {
+        log_error("Failed to open neatvnc server");
+        goto bail_aml;
+    }
+
+    tx->display = nvnc_display_new(0, 0);
+    if (!tx->display) {
+        log_error("Failed to create neatvnc display");
+        goto bail_server;
+    }
+
+    nvnc_add_display(tx->server, tx->display);
+    nvnc_set_name(tx->server, "Twin VNC Backend");
+    nvnc_set_pointer_fn(tx->server, _twin_vnc_pointer_event);
+    nvnc_set_new_client_fn(tx->server, _twin_vnc_new_client);
+    nvnc_set_userdata(tx->server, tx, NULL);
+
+    ctx->screen = twin_screen_create(width, height, _twin_vnc_put_begin,
+                                     _twin_vnc_put_span, ctx);
+    if (!ctx->screen)
+        goto bail_display;
+
+    tx->framebuffer = malloc(width * height * sizeof(uint32_t));
+    if (!tx->framebuffer) {
+        log_error("Failed to malloc framebuffer");
+        goto bail_screen;
+    }
+    memset(tx->framebuffer, 0xff, width * height * sizeof(uint32_t));
```

Comment:
Use `calloc` instead of the combination of `malloc` and `memset`. 
See https://gist.github.com/jadia/0806b7232773695b41f0e5c8a7fc9e0b